lib/cli/ui/prompt/interactive_options.rb

```diff
@@ -57,8 +65,9 @@ def call
             render_options
             process_input_until_redraw_required
             reset_position
+            clear_output if @displaying_metadata or not @answer.nil?
```

Comment:
Moving this inside the loop is only needed when the number of output lines goes down, and leads to unnecessary flashing of the output otherwise.  In fixing this, you'll need `CLI::UI.raw { print(ANSI.clear_to_end_of_line) }` in render options for the instruction/entered text line since it can get shorter.